src/vs/workbench/parts/terminal/electron-browser/windowsShellService.ts

```diff
@@ -0,0 +1,82 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import * as platform from 'vs/base/common/platform';
+import { IShellLaunchConfig } from 'vs/workbench/parts/terminal/common/terminal';
+import { TPromise } from 'vs/base/common/winjs.base';
+
+/** The amount of time to wait before getting the shell process name */
+const WAIT_FOR_SHELL_UPDATE = 100;
+
+export class WindowsShellService {
+	private _pidStack: number[];
+	private _processId: number;
+	private _shellLaunchConfig: IShellLaunchConfig;
```

Comment:
Let's pass in the executable instead (`_rootProcessExecutable`) as that's all this class cares about.